### PR TITLE
resolved dashboard not displaying contribution data issue

### PR DESCRIPTION
**Pull request by dhruvilpatel**

### CODE REVIEW

File: catalog/core/templates/dashboard.html [hunk 1]
```diff
@@ -73,7 +73,7 @@ <h3 class="panel-title">Recently Updated Publications</h3>
                                 {% endif %}
                                 </td>
                                 <td>{%  for key, value in contribution.items %}
-                                    {%  if key == pub.id %}
+                                    {%  if key == contribution_cache_prefix|addstr:pub.id %}
```
Comment (alee) [inline catalog/core/templates/dashboard.html]: I'm not a fan of this kind of munging. Is it possible to either change `contribution.items` to remove the cache prefix from the key or see if it's possible to use https://docs.djangoproject.com/en/2.0/ref/templates/builtins/#in-operator, e.g., `if pub.id in key` - I think that might cause false positives however:

 `'2170' in '12170' == True`

File: catalog/core/views.py [hunk 2]
```diff
@@ -192,6 +192,7 @@ def get_context_data(self, number_of_publications=25, **kwargs):
         context['recently_updated'] = recently_updated
         list = [CacheNames.CONTRIBUTION_DATA.value + str(pub.id) for pub in recently_updated]
```
Comment (alee) [inline catalog/core/views.py]: Don't shadow Python builtins, especially core builtins like `list`.

Can you explain the reason you are prefixing the publication id with CacheNames.CONTRIBUTION_DATA.value?

Comment (dhruvilpatel) [reply catalog/core/views.py]: We are using various default cachename as prefix inorder for better understanding and common globals name. for e.g. 'citation:contribution_data:pub_id-'. We have listed those names under citation.graphviz.globals.py. So inorder to access(get) contribution data for specific publication from the cache values we need to add those prefix values and append the specific publication id.

Comment (alee) [reply catalog/core/views.py]: Thanks for the explanation. So I think there should be a larger refactoring that has to go in here.

1. Use the dictionary as it was intended, with an actual key lookup - iterating across the keys and values of a dictionary comparing the dict key with some generated string (each time!) is way kludgy and inefficient
2. add a property like `pub.contribution_cache_key` which returns `CacheNames.CONTRIBUTION_DATA.value + self.id` and get rid of the over-and-over `add_str`

**Pull request merged**